README.md

```diff
@@ -1,229 +1,894 @@
+This information lets you identify in the context of which process and thread
+stack trace was captured.
+
+After this we see stack trace with some additional information. Note that
+functions on the top are calling functions below them, forming a top-down call
+chain.
+
+Each line in a stack trace corresponds to one kernel function call. Kernel
+function name is in the center (e.g,. `__x64_sys_bpf`, `map_create`,
+`array_map_alloc`, etc), plus a hexadecimal offset (in bytes) within that
+function, which allows to pinpoint exactly assembly instruction that was
+executed at that time. Functions that have '.' (dot) in front of them (e.g., `.
+map_create` and `.  find_and_alloc_map`) are inlined functions and they don't
+really have their own stack frame in the stack trace, but thanks to available
+DWARF debug information `retsnoop` managed to recover this information and
+present it as a `__sys_bpf` -> `map_create` -> `find_and_alloc_map` call chain,
+which is what is logically happening according to C source code.
+
+To the right of most functions we can see source code line location that was
+executing at the time of capturing stack trace. `kernel/bpf/syscall.c:1106:8`
+(next to `. map_create`) points out that C source code line #1106 in
+`kernel/bpf/syscall.c` file was the one being executed at the time of
+`find_and_alloc_map` -> `array_map_alloc` -> `bpf_map_alloc_percpu` call chain
+executing inside `map_create`. In source code corresponding to the kernel I've
+run this maps correctly to:
+
+```c
+  1105        /* find map type and init map: hashtable vs rbtree vs bloom vs ... */
+  1106        map = find_and_alloc_map(attr);
+  1107        if (IS_ERR(map))
+  1108                return PTR_ERR(map);
+```
+
+To the left of each function that was specified through `-e` and `-a` arguments
+you'll also see two more pieces of information: function latency (e.g., 73
+microseconds for `__x64_sys_bpf`) and it's return result, which could be one of
+`-Exxx` error codes, `NULL`, `true`/`false`, some valid address like
+`0xffffc90000243000`, some other integer value, or even `void`; `retsnoop` will
+do its best to represent captured return value as meaningfully as possible,
+based on captured return value and function's type signature, if available. In
+our case, we see that innermost call to `bpf_map_alloc_percpu` returned `NULL`,
+which the got translated to `-ENOMEM` error and bubbled up all the way through
+`__x64_sys_bpf` and eventually got returned to user-space as a result of
+`bpf()` syscall.
+
+There could be few extra markers like '!', '*', etc, at the very left side of a
+line, but we won't go into detail on what they mean. Usually it's pretty
+harmless to ignore them.
+
+So, in general, what did we achieve and what can we understand from
+`retsnoop`'s output? With a single invocation of `retsnoop` with few arguments
+we managed to trace `bpf()` syscall all the way down to `bpf_map_alloc_percpu`
```

Comment:
s/trace `bpf()` syscall all the way down to/trace the `bpf()` syscall all the way down to the